cmd/utils/config.go

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package utils
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path"
+
+	"github.com/spf13/viper"
+)
+
+const (
+	currentInstanceKey = "current-instance"
+	configFilename     = "config.yaml"
+)
+
+var config = viper.New()
+
+func init() {
+	config.SetConfigFile(path.Join(configDir(), configFilename))
+}
+
+// ShowConfig prints the configuration to stdout
+func ShowConfig() error {
+	cfg, err := os.ReadFile(path.Join(configDir(), configFilename))
+	if err != nil && !errors.Is(err, fs.ErrNotExist) {
+		return fmt.Errorf("failed to read config file: %w", err)
+	}
+	fmt.Fprintf(os.Stdout, "%s", cfg)
+	return nil
+}
+
+// UseInstanceConfig sets the current instance to use in the configuration
+func UseInstanceConfig(instance string) error {
+	if err := loadConfig(); err != nil {
+		return fmt.Errorf("failed to load config: %w", err)
+	}
+	if !config.IsSet("instances." + instance) {
+		return fmt.Errorf("instance %q not found", instance)
+	}
+	config.Set(currentInstanceKey, instance)
+	return config.WriteConfig()
+}
+
+// SetInstanceConfig sets the instance configuration based on the provided flags
+func SetInstanceConfig(instance string) error {
+	if err := loadConfig(); err != nil {
+		return fmt.Errorf("failed to load config: %w", err)
+	}
+	// delete the instance if it already exists
+	if config.IsSet("instances." + instance) {
+		if err := deleteInstanceConfig(instance); err != nil {
+			return fmt.Errorf("failed to delete instance config: %w", err)
+		}
+	}
+
+	if node != "" {
+		config.Set("instances."+instance+".node", node)
+	} else if resourceID != "" {
+		config.Set("instances."+instance+".id", resourceID)
+	} else {
+		config.Set("instances."+instance+".subscription", subscriptionID)
+		config.Set("instances."+instance+".node-resource-group", nodeResourceGroup)
+		config.Set("instances."+instance+".vmss", vmScaleSet)
+		config.Set("instances."+instance+".instance-id", vmssInstanceID)
+	}
+
+	return config.WriteConfig()
```

Comment:
I removed the config file and test:

```bash
$ ./kubectl-az config set-instance "myInstance1" --subscription "mySub" --node-resource-group "myNRG" --vmss "myVMSS" --instance-id "myInstanceID"
Error: open /home/jose/.kubectl-az/config.yaml: no such file or directory
```

I suggest to add a wrapper to easily identify where the error come from:
```suggestion
	if err := config.WriteConfig(); err != nil {
		return fmt.Errorf("writing config: %w", err)
	}
	return nil
```

Reply:
Thanks. Let me do it. I realized we need to create the config dir if it doesn't exists for the first time. I will add that!